forecasttools/idata_w_dates_to_df.py

```diff
@@ -7,80 +7,226 @@
 scoringutils ready dataframes
 """
 
-from datetime import datetime
+from datetime import date, datetime, timedelta
 
 import arviz as az
 import numpy as np
 import polars as pl
+import xarray as xr
 
+import forecasttools
 
-def add_dates_as_coords_to_idata(
-    idata_wo_dates: az.InferenceData,
-    group_dim_dict: dict[str, str],
-    start_date_iso: str,
+
+def generate_time_range_for_dim(
+    start_time_as_dt: datetime | date,
+    variable_data: xr.DataArray,
+    dimension: str,
+    time_step: timedelta,
+) -> list[date] | list[datetime]:
+    """
+    Generates a range of times based on the
+    start date, time step, and variable's
+    dimension size. A range of dates is
+    generated if the start date is a date and
+    the time step is in days. A range of times
+    is generated if the start date is a time
+    and/or the time step is a time.
+    """
+
+    # get the size of the dimension
+    interval_size = variable_data.sizes[dimension]
+
+    # check if time_step is in full days or
+    # weeks
+    is_full_days_or_weeks = (
+        time_step.total_seconds() % timedelta(days=1).total_seconds() == 0.0
```

Comment:
I think I prefer a helper function, and one that avoids float comparison with 0. Something like

```python
def is_integer_days(td: datetime.timedelta):
    if td.resolution > datetime.timedelta(days=1):
        raise ValueError("Cannot check whether a timedelta "
                         "represents an integer number of days "
                         "for a timedelta with resolution greater than 1 day. "
                         f"Got {td.resolution}")
    return abs(td -  td.days) < t.resolution
```

Reply:
I am in the process of adding something like this.